src/FixedLocalTupleList.cpp

```diff
@@ -0,0 +1,255 @@
+        }
+        
+        tuple::iterator it = pids.begin();
+        vp = storage->lookupRealParticle(*it);
+        if (!vp) {// Particle does not exist here, return false
+            returnVal = false;
+        } else {
+            pidK = *it; // first pid is key
+            for (++it; it != pids.end(); ++it) {
+                at = storage->lookupLocalParticle(*it);
+                if (!at) { // Particle does not exist here, return false
+                    std::stringstream msg;
+                    msg << "ERROR: particle " << *it << " not found \n";
+                    err.setException(msg.str());
+                    returnVal = false;
+                    break;
+                }
+                tmp.push_back(at);
+                pidstmp.push_back(*it); // pidK is not in this vector
+            }
+        }
+        err.checkException();
+	
+        if (returnVal) {
+            this->add(vp, tmp); // add to TupleList
+            globalTuples.insert(make_pair(pidK, pidstmp));
+	    LOG4ESPP_INFO(theLogger, "added fixed tuple to global tuples");
+        }
+	
+        tmp.clear();
+        pids.clear();
+        pidstmp.clear();
+	
+
+        return returnVal;
+    }
+
+    python::list FixedLocalTupleList::getTuples() {
+        python::list tuple;
+        python::list alltuples;
+        for (GlobalTuples::iterator it = globalTuples.begin();
+                it != globalTuples.end(); it++) {
+            python::list tuple;
+            tuple.append((*it).first); // key is also part of the tuple!
+            for (tuple::iterator it2=(*it).second.begin(); it2 !=(*it).second.end(); it2++){
+                tuple.append(*it2);
+            }
+            alltuples.append(tuple);
+        }
+	
+        return alltuples;
+    }
+    
+    void FixedLocalTupleList::
+    beforeSendParticles(ParticleList& pl, OutBuffer& buf) {
+        std::vector<longint> toSend;
+        // loop over the particle list
+        for (ParticleList::Iterator pit(pl); pit.isValid(); ++pit) {
+            longint pidK = pit->id();
+            LOG4ESPP_DEBUG(theLogger, "send particle with pid " << pidK << ", find tuples");
+	    
+            // find particle that involves this particle id
+            GlobalTuples::const_iterator it = globalTuples.find(pidK);
+            if (it != globalTuples.end()) {
+		int s = it->second.size();
+		toSend.reserve(toSend.size() + s + 2);
+                // first write the pid of the first particle
+		toSend.push_back(pidK);
+                // write the size of the vector
+		toSend.push_back(s);
+
+                // iterate through vector and add pids
+                for (tuple::const_iterator it2 = it->second.begin();
+                        it2 != it->second.end(); ++it2) {
+                    toSend.push_back(*it2);
+                }
+                // delete this pid from the global list
+                globalTuples.erase(pidK);
+		//std::cout << "Erase pid " << pidK << std::endl;
```

Comment:
Please for this kind of message use logger instead of commented code, perhaps LOG4ESPP_DEBUG here?